core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -861,6 +835,8 @@ private boolean deleteIfFound(Term uid, long currentVersion, boolean deleted, Ve
             found = false;
         } else {
             // we deleted a currently existing document
+            // any exception that comes from this is a either an ACE or a fatal exception there can't be any document failures coming
```

Comment:
this is difficult since I need to throw a VirtualMachineError to trigger this and then all our test will go nuts I think we gotta trust this here